packages/pancake-uikit/src/widgets/WalletModal/ConnectModal.tsx

```diff
@@ -1,40 +1,95 @@
+  border-bottom: 1px solid ${({ theme }) => theme.colors.cardBorder};
 `;
 
-const ConnectModal: React.FC<Props> = ({ login, onDismiss = () => null }) => (
-  <Modal title="Connect to a wallet" onDismiss={onDismiss}>
-    {config.map((entry, index) => (
-      <WalletCard
-        key={entry.title}
-        login={login}
-        walletConfig={entry}
-        onDismiss={onDismiss}
-        mb={index < config.length - 1 ? "8px" : "0"}
-      />
-    ))}
-    <HelpLink href="https://docs.pancakeswap.finance/get-started/connection-guide" external>
-      <HelpIcon color="primary" mr="6px" />
-      Learn how to connect
-    </HelpLink>
-  </Modal>
-);
+/**
+ * Checks local storage if we have saved the last wallet the user connected with
+ * If we find something we put it at the top of the list
+ *
+ * @returns sorted config
+ */
+const getPreferredConfig = (walletConfig: Config[]) => {
+  const preferredWalletName = localStorage.getItem(walletLocalStorageKey);
+  const sortedConfig = walletConfig.sort((a: Config, b: Config) => a.priority - b.priority);
+
+  if (!preferredWalletName) {
+    return sortedConfig;
+  }
+
+  const preferredWallet = sortedConfig.find((sortedWalletConfig) => sortedWalletConfig.title === preferredWalletName);
+
+  if (!preferredWallet) {
+    return sortedConfig;
+  }
+
+  return [
+    preferredWallet,
+    ...sortedConfig.filter((sortedWalletConfig) => sortedWalletConfig.title !== preferredWalletName),
+  ];
+};
+
+const ConnectModal: React.FC<Props> = ({ login, onDismiss = () => null, displayCount = 3 }) => {
+  const [showMore, setShowMore] = useState(false);
+  const theme = useTheme();
+  const sortedConfig = getPreferredConfig(config);
+  const displayListConfig = showMore ? sortedConfig : sortedConfig.slice(0, displayCount);
+
+  return (
+    <ModalContainer minWidth="320px">
+      <ModalHeader background={getThemeValue("colors.gradients.bubblegum")(theme)}>
+        <ModalTitle>
+          <Heading>Connect Wallet</Heading>
+        </ModalTitle>
+        <ModalCloseButton onDismiss={onDismiss} />
+      </ModalHeader>
+      <ModalBody width={["320px", null, "340px"]}>
+        <WalletWrapper py="24px" maxHeight="453px" overflowY="auto">
+          <Grid gridTemplateColumns="1fr 1fr">
+            {displayListConfig.map((wallet) => (
+              <Box key={wallet.title}>
+                <WalletCard walletConfig={wallet} login={login} onDismiss={onDismiss} />
+              </Box>
+            ))}
+            {!showMore && <MoreWalletCard onClick={() => setShowMore(true)} />}
+          </Grid>
+        </WalletWrapper>
+        <Box p="24px">
+          <Text textAlign="center" color="textSubtle" as="p" mb="16px">
+            Haven&#39;t got a crypto wallet yet?
+          </Text>
+          <Button
+            as="a"
+            href="https://docs.pancakeswap.finance/get-started/connection-guide"
+            variant="subtle"
+            width="100%"
+            target="_blank"
```

Comment:
you can use `as={Link}` and the `external` props

Reply:
I tried that initially but it messages with the styling. You cannot seem to apply other styles like `width="100%"` and `display="block"`.

I did however, use the `getExternalProps` helper.